### DISCUSSION TITLE
Error: This version of libvalhalla was not built with CURL support

### DISCUSSION THREAD

**Discussion by girishyadav20** [gis-ops/pyvalhalla#40] [Q&A]
Hi,

I am using pyValhalla. When I set tile_url config parameter, it throws the error: "**This version of libvalhalla was not built with CURL support**".

How to fix this issue? How to include curl with pyValhalla?

Thanks!
Girish

**Comment by baileyheading** [comment]
If you want to use CURL with pyvalhalla, I would recommend using something like FastAPI in python and making endpoints that access it. However if you're just using Valhalla and not requiring python, then Valhalla in c++ has an api server you can host and that'd be the better option

**Reply by girishyadav20** [reply]
Thanks for the response! However, I am not creating a server with pyvalhalla. I am trying to set tile_url valhalla configuration so that valhalla can download the required road graph tile as needed from the tile server (which I have already setup). 

When I do something like this (As mentioned in the Help text for Mjolnir.tile_url config):

config["mjolnir"]["tile_url"] = "https://valhalla-tile-service/tiles{tilePath}"
actor = Actor(config)

It throws the error "This version of libvalhalla was not built with CURL support".

**Reply by baileyheading** [reply]
Ok, I didn't know Valhalla had support for that honestly. Which documentation are you referring to? (Could you send a link to it?). If it is supported, as per the error message it was probably not built for in this version used here. In which case fastest fix is to build the static libraries yourself and swap them into PyValhalla (or make and use the python bindings yourself on the Valhalla repo with the build settings you want)

**Reply by girishyadav20** [reply]
Please check the https://github.com/gis-ops/pyvalhalla/blob/master/valhalla/valhalla_build_config.py file. Inside look for "help_text" and in the dictionary look for "mjolnir"."tile_url".

**Reply by girishyadav20** [reply]
"tile_url": "Http location to read tiles from if they are not found in the tile_dir, e.g.: http://your_valhalla_tile_server_host:8000/some/Optional/path/{tilePath}?some=Optional&query=params. Valhalla will look for the {tilePath} portion of the url and fill this out with a given tile path when it make a request for that tile"
